lib/gambit/list/list.scm

```diff
@@ -607,22 +607,26 @@
                 (macro-check-list x 2 (assoc obj lst c)
                   #f))))))))
 
-(define-prim (##remq elem lst)
-
-  (include "~~lib/gambit/prim/prim#.scm") ;; map fx+ to ##fx+, etc
-
-  (define (rem elem lst)
-    (if (pair? lst)
-        (let* ((lst2 (rem elem (cdr lst)))
-               (x (car lst)))
-          (if (eq? x elem)
-              lst2
-              (if (eq? lst2 (cdr lst))
-                  lst
-                  (cons x lst2))))
-        '()))
-
-  (rem elem lst))
+(define-macro (##filter-macro lst var test-expr)
+  `(let recurse ((lst ,lst))
+     (if (pair? lst)
```

Comment:
I tend to not use named `let` for recursions and prefer to do `(let () (define (recurse ...) ...) (recurse ...))`.  I also avoid the name `recurse` that sounds like repeated cursing...